### PR TITLE
Document create/drop catalog

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Document create/drop catalog with a lot of warnings about known issues.

This adds 3 pages:
* CREATE CATALOG
* DROP CATALOG
* Catalog management properties

There are more existing and undocumented catalog management properties, like `catalog.config-dir`, that can be added to the new properties page later.

I decided to repeat the same warnings on multiple pages, since they're very important.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by nineinchnick** [comment]
Some screenshots to test if the warnings are scary enough :-)
image

image

image

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (nineinchnick) [inline docs/src/main/sphinx/sql/create-catalog.md:14]: @dain is the new catalog available for new queries immediately after this query finishes, or is it loaded asynchronously? Either way, is it worth mentioning here?

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [inline docs/src/main/sphinx/admin/properties-catalog.md:35]: I just noticed I missed `catalog.prune.enabled` but I'm not sure what's the use case for disabling it.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [inline docs/src/main/sphinx/admin/properties-catalog.md:14]: I guess this is only true if `catalog.store=file`, which is the default, but since it's not documented, maybe we can update this when documenting `catalog.store` and other properties?

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (dain) [inline docs/src/main/sphinx/admin/properties-catalog.md:25]: I'd mention S3, GCS, and Azure

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (dain) [reply docs/src/main/sphinx/admin/properties-catalog.md:35]: I'm not sure why I added that.  Maybe it made testing easier.  Anyway, I'm not sure we need to document the enable

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (dain) [reply docs/src/main/sphinx/sql/create-catalog.md:14]: It is immediately available when the command returns.  Actually constructing the connector is pretty fast.  When you run queries for the first time you can get some slowness as connections are established, and some connectors cache information and the caches would be empty

**Review by dain — COMMENTED** [review]
Looks good to me

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:4]: no empty line needed between title and anchor

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:14]: We should document these properties now as well

Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:14]:
```suggestion
catalogs using these SQL commands adds and removes catalog
```

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:22]: Do we really still need this. I thought this was a problem from the Hadoop dependencies and those are all gone. And even if .. its only applicable for LOTS of create/drop cycles .. that should be rare.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:15]: We also need to clarify that it requires read access to the filesystem on the coordinator and workers by the process running Trino.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:18]: I dont think the syntax changes and the security implications are necessarily related.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:15]: Also .. what happens if a new node is added via cluster scaling for example on k8s .. will it somehow get the catalog files created as necessary or not?

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:31]:
```suggestion
The complete `CREATE CATALOG` query is logged, and visible in the {doc}`Web
```

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:42]: we should add info on where they are dropped .. on coordinator and all workers..

File: docs/src/main/sphinx/language/sql-support.md
Comment (mosabua) [inline docs/src/main/sphinx/language/sql-support.md:103]: I guess there is no ALTER or SHOW CREATE

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:14]: Do we really need to say that its ready? If anything we should say that it creates the necessary info on all nodes in the cluster and therefore might take some time.. if that is indeed the case (I mean how long it takes)

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:17]:
```suggestion
The optional `WITH` clause is used to set properties
```

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:19]: Add something about valid properties depend on the connector .. refer to the docs for those.

Also how does it behave if you create invalid config? Is there any validation or does it just create them and then you have to test if it actually worked afterwards.

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/sql/create-catalog.md:24]: We should add an example that does that

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:32]: Add a super simple example that does not use WITH properties first .. with the tpch connector.

Then add a simple one with the memory connector (use catalog name "brain" maybe like in the book) and the simple config like on that connectors docs for memory allocation.

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:34]:
```suggestion
Create a new catalog called `example` using the PostgreSQL conenctor:
```

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:37]:
```suggestion
CREATE CATALOG example USING postgresql
```

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:38]: Expand to use env secrets with username and password and maybe some other postgresql connector config. And explain

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/drop-catalog.md:15]: Again .. not sure we need this

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/drop-catalog.md:25]: Does it need write access to the file system? How does it work if not ..e .g. with k8s?

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/drop-catalog.md:28]:
```suggestion
Drop the catalog `example`:
```

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/drop-catalog.md:31]:
```suggestion
DROP CATALOG example;
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/admin/properties-catalog.md:19]: I was testing this a few days ago and there was a 10% increase in non-heap memory usage after creating and dropping a hive catalog 15 times.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/admin/properties-catalog.md]: I tested this locally in containers and workers get all the catalogs when joining the cluster (they're able to execute queries), but they don't write the files to disk. I updated relevant parts of the docs.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/admin/properties-catalog.md:15]: I'd leave this, we can always remove this later when we figure out how to handle sensitive values even if there are no syntax changes.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/admin/properties-catalog.md]: This is not relevant here, but I explained that in the description of other properties.

File: docs/src/main/sphinx/language/sql-support.md
Comment (nineinchnick) [reply docs/src/main/sphinx/language/sql-support.md:103]: Not yet

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/sql/create-catalog.md:14]: This is correct, you can run a query using this catalog _immediately_ after this one completes, and it'll start running, just might be a bit slower.

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/sql/create-catalog.md]: Added all of this and a long list of how properties are validated, thanks!

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/sql/create-catalog.md:43]: Done!

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/sql/drop-catalog.md:15]: Explained in the other comment.

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/sql/drop-catalog.md:25]: In the Trino Docker image, the `/etc/trino/catalog` dir is already writeable by the user running the Trino server. I explained all the permission details in the properties page.

**Comment by nineinchnick** [comment]
@mosabua great review, thanks! I added a lot more info as a separate commit. This just proves how complex this can get and how important it is to have extensive docs :-)

**Comment by nineinchnick** [comment]
For k8s, I'll check later if it's easy to mount a persistent volume at /etc/trino/catalog when using the trino helm chart and if we need to document how to do that.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:15]: ok

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:19]: Lets leave it for now .. but we have to keep an eye on it and remove it when it no longer applies

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:6]: All these links that use 

```
{ref}`prop-type-string`
```

should be changed to use the shorter markdown syntax

```
[](prop-type-string)
```

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:22]: All these links should change in format to 

```
[](/connector/hive)
```

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-catalog.md:29]: change link syntax

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/sql/create-catalog.md:14]: I know .. but do we really need to say that .. its like saying you can look at the results of a SQL query when the results are displayed..

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:19]:
```suggestion
special characters, like `-`. Refer to the [connectors documentation](/connector)
```

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:23]:
```suggestion
The query fail in the following circumstances:

```

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:39]:
```suggestion
This command requires the [catalog management type](/admin/properties-catalog) to
```

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:57]: Note the required use of " for quoting the property name, since the dash characters in the property name are invalid in SQL. The value of 128MB must be quoted with standard '. sin and the required ' to

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-catalog.md:71]:
```suggestion
environmental variables are set as [secrets](/security/secrets) on the coordinator node.
```

or some similar link to the secrets docs

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (mosabua) [inline docs/src/main/sphinx/sql/drop-catalog.md:17]: change links

**Review by mosabua — CHANGES_REQUESTED** [review]

**Comment by mosabua** [comment]
More comments .. very close. We should add some note that with dynamic mode catalogs are managed by the coordinator automatically so it also fully works for scaling clusters .. super cool!

**Comment by nineinchnick** [comment]
All comments applied. I tested this locally and all links look good. I've also run Vale this time and there are no issues.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (hashhar) [inline docs/src/main/sphinx/admin/properties-catalog.md:83]: how is this achieved? I don't see this happenning on the filesystem level. Maybe we mean to say that enabling this disables `CREATE CATALOG` and `DROP CATALOG`.

Additionally with catalog store set to file and mode set to dynamic Trino won't (AFAIK) "reload" changed files or newly added file to catalog directory by hand - this needs to be called out IMO.

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (hashhar) [inline docs/src/main/sphinx/sql/create-catalog.md:30]: does the variable not need to exist on worker nodes too? Or do workers get the secret value "inlined" into the config (hence no longer being secrets on the worker)?

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (hashhar) [inline docs/src/main/sphinx/sql/drop-catalog.md:16]:
```suggestion
that uses such connector. This includes all connectors, but is not limited to, that can read data from
```

or something to that effect

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/admin/properties-catalog.md:83]: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/connector/FileCatalogStore.java#L131

Comment (nineinchnick) [reply docs/src/main/sphinx/admin/properties-catalog.md:83]: Why it needs to be called out? There has been no reload mechanism in Trino even with the pre-existing static catalog management.

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/sql/create-catalog.md:30]: It does not need to exist on worker nodes. I assume workers receive already resolved values from the coordinator and store them in memory.

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (nineinchnick) [reply docs/src/main/sphinx/sql/drop-catalog.md:16]: The first sentence in this warning says we haven't tested all connectors, isn't that enough?

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (hashhar) [reply docs/src/main/sphinx/admin/properties-catalog.md:83]: > https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/connector/FileCatalogStore.java#L131

Yes, so  it disables `CREATE` and `DROP` only - it doesn't disable "existing catalog property files cannot be deleted or modified and new catalog property files cannot be created".

Though I'm interested about when `dynamic` + `read-only` makes sense. It's probably a question to @dain.

File: docs/src/main/sphinx/sql/drop-catalog.md
Comment (hashhar) [reply docs/src/main/sphinx/sql/drop-catalog.md:16]: people read too much into lists. does it hurt being explicit?

Comment (nineinchnick) [reply docs/src/main/sphinx/sql/drop-catalog.md:16]: I don't have a strong opinion on this, except for not using too long sentences :-) I'd defer to @mosabua

Comment (mosabua) [reply docs/src/main/sphinx/sql/drop-catalog.md:16]: From what I know the memory leakage is coming from the HDFS/Hadoop libraries only - so only the object storage connectors are affected. Once those are removed as dependencies (very soon) the whole note will be removed.

Comment (mosabua) [reply docs/src/main/sphinx/sql/drop-catalog.md:16]: Lets not have this hedging language

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/sql/create-catalog.md:30]: Whats what I understand from @dain

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:83]: Lets reword this accordingly

Comment (nineinchnick) [reply docs/src/main/sphinx/admin/properties-catalog.md:83]: I don't understand the difference. @mosabua you can push a change directly to this branch.

Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:83]: Done

**Review by mosabua — APPROVED** [review]
Looks great now. Users will get excited to see these docs and start testing it out. Once the Hadoop libraries are removed as dependencies we can drop the warning about the object storage connectors.

**Comment by mosabua** [comment]
I squashed commits and did some minor corrections btw.

File: docs/src/main/sphinx/admin/properties-catalog.md
Comment (nineinchnick) [inline docs/src/main/sphinx/admin/properties-catalog.md:83]: There's a typo `s/now/no`

Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-catalog.md:83]: Dang.. can you fix that in your follow up PR maybe?

File: docs/src/main/sphinx/sql/create-catalog.md
Comment (hashhar) [reply docs/src/main/sphinx/sql/create-catalog.md:31]: https://github.com/trinodb/trino/pull/19883

**Pull request merged**
